cli/command/image/build/context_detect.go

```diff
@@ -0,0 +1,39 @@
+package build
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docker/docker/builder/remotecontext/urlutil"
+)
+
+// ContextType describes the type (source) of build-context specified.
+type ContextType string
+
+const (
+	ContextTypeStdIn  ContextType = "stdin"  // ContextTypeStdIn indicates that the build-context is a TAR archive passed through STDIN.
```

Comment:
nit: IMO `StdIn` looks weird and isn't consistent with the general naming (like `os.Stdin`).

```suggestion
	ContextTypeStdin  ContextType = "stdin"  // ContextTypeStdin indicates that the build-context is a TAR archive passed through STDIN.
```

(just an opinion though)

Reply:
Ah, yeah, can change; I always read its as standard-in / standard-out, standard-err, not as a single word, but Golang uses `os.Stdin`, so it makes sense to follow that.

Let me update.

Reply:
done! Will merge when CI finishes.